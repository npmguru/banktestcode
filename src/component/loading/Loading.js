import React, { useEffect,useState } from 'react';
import loading from '../../images/loading.svg';
import logo from '../../images/logo.png';
import { Redirect } from "react-router-dom";
import './Loading.css';

const Loading = () => {
  const [isRedirect, setRedirect] = useState(false)
  useEffect(() => { 
        setTimeout(() => { setRedirect(true) }, 3000);
   });
  
  return (
    <div className="Loading">
     {isRedirect ? (<Redirect to="/login" />) : (
      <div className="Loading-Content">
        <img src={logo} width='200' height='200' alt='logo'  />
        <img src={loading} alt="Loading...."  />
      </div>
     )}

    </div>
  );
}

export default Loading;
