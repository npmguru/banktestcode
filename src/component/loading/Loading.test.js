import React from 'react';
import { shallow } from 'enzyme';
import Loading from './Loading';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Loading /> with no props', () => {

  const container = shallow(<Loading />);

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  
});