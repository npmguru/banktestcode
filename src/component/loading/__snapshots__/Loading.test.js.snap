// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> with no props should match the snapshot 1`] = `"<div class=\\"Loading\\"><div class=\\"Loading-Content\\"><img src=\\"logo.png\\" width=\\"200\\" height=\\"200\\" alt=\\"logo\\"/><img src=\\"loading.svg\\" alt=\\"Loading....\\"/></div></div>"`;
